openssl/src/ssl/mod.rs

```diff
@@ -652,6 +654,21 @@ impl SslContextBuilder {
         }
     }
 
+    /// Sets a custom X509Store for verifying peer certificates.
+    ///
+    /// Requires the `v102` feature and OpenSSL 1.0.2, or the `v110` feature and OpenSSL 1.1.0.
+    #[cfg(any(all(feature = "v102", ossl102), all(feature = "v110", ossl110)))]
+    pub fn set_verify_cert_store(&mut self, cert_store: X509Store) -> Result<(), ErrorStack> {
+        unsafe {
+            // set0 will free, set1 increments, and then requires a free
+            let ptr = cert_store.as_ptr();
+            let result = cvt(ffi::SSL_CTX_set0_verify_cert_store(self.as_ptr(), ptr) as c_int).map(|_|());
+            
+            mem::forget(cert_store);
```

Comment:
This is still unconditionally forgetting the store, right? You should just be able to `try!` on the call above, forget, and then return `Ok(())`.